core/src/server/rpc_module.rs

```diff
@@ -682,34 +627,30 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	/// use jsonrpsee_core::Error;
 	///
 	/// let mut ctx = RpcModule::new(99_usize);
-	/// ctx.register_subscription("sub", "notif_name", "unsub", |params, mut sink, ctx| {
-	///     let x = match params.one::<usize>() {
-	///         Ok(x) => x,
-	///         Err(e) => {
-	///             let err: Error = e.into();
-	///             sink.reject(err);
-	///             return Ok(());
-	///         }
-	///     };
-	///     // Sink is accepted on the first `send` call.
-	///     std::thread::spawn(move || {
-	///         let sum = x + (*ctx);
-	///         let _ = sink.send(&sum);
-	///     });
+	/// ctx.register_subscription("sub", "notif_name", "unsub", |params, pending, ctx| async move {
```

Comment:
The above comment may need updating; `callback` to mention `PendingSubscriptionSink` again :)